integration/package-operator/zzz_upgrade_test.go

```diff
@@ -0,0 +1,213 @@
+package packageoperator
+
+import (
+	"bytes"
+	"context"
+	goerrors "errors"
+	"fmt"
+	"io"
+	"net/http"
+	"path/filepath"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/require"
+
+	batchv1 "k8s.io/api/batch/v1"
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+
+	corev1alpha1 "package-operator.run/apis/core/v1alpha1"
+
+	"github.com/go-logr/logr"
+	"github.com/go-logr/logr/testr"
+	"github.com/mt-sre/devkube/dev"
+
+	"k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+func TestUpgrade(t *testing.T) {
+	ctx := logr.NewContext(context.Background(), testr.New(t))
+
+	require.NoError(t, deleteExistingPKO(ctx))
+
+	log := logr.FromContextOrDiscard(ctx)
+	pkg := &corev1alpha1.ClusterPackage{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "package-operator",
+			Namespace: "package-operator-system",
+		},
+	}
+
+	log.Info("Installing latest released PKO")
+	require.NoError(t, createAndWaitFromHTTP(ctx, []string{LatestSelfBootstrapJobURL}))
+
+	jobList := &batchv1.JobList{}
+	require.NoError(t, Client.List(
+		ctx, jobList,
+		client.InNamespace("package-operator-system"),
+	))
+	for i := range jobList.Items {
+		require.NoError(t, Waiter.WaitToBeGone(ctx, &jobList.Items[i], func(obj client.Object) (done bool, err error) { return false, nil }, dev.WithTimeout(6*time.Minute)))
+	}
+
+	require.NoError(t, Waiter.WaitForCondition(ctx, pkg, corev1alpha1.PackageAvailable, metav1.ConditionTrue, dev.WithTimeout(5*time.Minute)))
+	log.Info("Latest released PKO is now available")
+
+	cosList := &corev1alpha1.ClusterObjectSetList{}
+	require.NoError(t, Client.List(ctx, cosList, client.MatchingLabels{"package-operator.run/object-deployment": "package-operator"}))
```

Comment:
I think we have variable definitions for the label name (and maybe value too?) somewhere in the codebase